homeassistant/components/solarlog/sensor.py

```diff
@@ -86,36 +86,40 @@ class SolarLogInverterSensorEntityDescription(SensorEntityDescription):
         translation_key="yield_day",
         native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
         device_class=SensorDeviceClass.ENERGY,
-        value_fn=lambda data: round(data.yield_day / 1000, 3),
+        suggested_display_precision=3,
+        value_fn=lambda data: data.yield_day / 1000,
```

Comment:
You can pull these even further, setting native UoM to Watt hour, and suggested_unit_of_measurement to kWh. This saves another conversion